### PR TITLE
DSDEGP-2525 Create a new integration "testkit" for use in other projects

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

For use in Reaper to start, but will hopefully be useful elsewhere too as more things start talking to Clio.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-integration-testkit/src/main/scala/org/broadinstitute/clio/integrationtest/ClioDockerComposeContainer.scala [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    logDir.createDirectories()
+
+    // Extract logging config to mount.
+    val _ = (
+      ClioDockerComposeContainer.extractClioLogback(logDir),
+      ClioDockerComposeContainer.extractEsLog4j(logDir)
+    )
+
+    // Make sure target log files are writeable.
+    (clioLog +: esLogs).foreach { log =>
```
Comment (tlangs) [inline clio-integration-testkit/src/main/scala/org/broadinstitute/clio/integrationtest/ClioDockerComposeContainer.scala:142]: Why do we need to do this?

File: project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala [hunk 2]
```diff
@@ -169,6 +170,12 @@ object Dependencies {
   val DataaccessModelDependencies: Seq[ModuleID] =
     DataaccessModelMainDependencies ++ DataaccessModelTestDependencies
 
+  val IntegrationTestkitDependencies: Seq[ModuleID] = Seq(
+    "com.dimafeng" %% "testcontainers-scala" % TestContainersScalaVersion,
+    "com.github.pathikrit" %% "better-files" % BetterFilesVersion,
```
Comment (tlangs) [inline project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala:175]: Don't we use `better-files` all over the place? why do we need to explicitly declare it for integration tests?

**Review by tlangs — APPROVED** [review]
I love getting rid of TestContainers!

File: clio-integration-testkit/src/main/scala/org/broadinstitute/clio/integrationtest/ClioDockerComposeContainer.scala (hunk 1 above)
Comment (danxmoran) [reply clio-integration-testkit/src/main/scala/org/broadinstitute/clio/integrationtest/ClioDockerComposeContainer.scala:142]: If the files don't exist, Docker mounts the paths as directories and the servers fall over trying to write logs. If the files exist but aren't writeable, Docker doesn't chmod them when it mounts so the servers still fall over trying to write.

File: project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala (hunk 2 above)
Comment (danxmoran) [reply project/src/main/scala/org/broadinstitute/clio/sbt/Dependencies.scala:175]: The new module doesn't depend on any of the other Clio subprojects, so it doesn't pull in the deps transitively.

**Pull request merged**